backend/inspirehep/records/api/literature.py

```diff
@@ -89,6 +89,53 @@ def create(
             record.push_authors_phonetic_blocks_to_redis()
             return record
 
+    @classmethod
+    def get_es_linked_records_from_dict_field(cls, data, path):
+        """Return the generator of linked records from specified path for records in ES.
+
+        Args:
+            data (dict): data from which records should be extracted
+            path (str): the path of the linked records.
+        Yields:
+            InspireRecord: the linked records.
+        Examples:
+            >>> data = {
+                'references': [
+                    {
+                        'record': {
+                            '$ref': 'http://localhost/literature/1'
+                        }
+                    }
+                ]
+            }
+            >>>  records = LiteratureRecord.get_linked_records_from_field(
+                data, "references.record")
+            >>> list(records)
+            [
+                {
+                    '$schema': 'http://localhost/schemas/records/hep.json'
+                    'control_number': 1,
+                    'authors': [(...)],
+                    'arxiv_eprints':  [(...)],
+                    'collaborations': [(...)],
+                }
+            ]
+        """
+        pids = cls._get_linked_pids_from_field(data, path)
+        if pids:
+            results = LiteratureSearch.get_records_by_pids(pids)
+            for result in results:
+                try:
+                    rec_data = json.loads(result._ui_display)
```

Comment:
how do you know the data you're interested in is actually present in `_ui_display`? and what's the point of using that if you're deserializing it anyway? wouldn't it make more sense to have a configurable `_source`?

Reply:
We don't have everything in `_ui_display`, we are missing  `arxiv_eprints`, `collaborations`, `dois` and `misc` field from there, this is why we are merging deserialized `_ui_display` with this missing fields in line 135. I preffer to use `_ui_data` as this is only place where we have limited number of authors there. Big number of authors was the main reason why loading of references was so slow before.